src/metabase/api/automagic_dashboards.clj

```diff
@@ -272,22 +272,24 @@
 (api/defendpoint GET "/:entity/:entity-id-or-query/rule/:prefix/:dashboard-template"
   "Return an automagic dashboard for entity `entity` with id `id` using dashboard-template `dashboard-template`."
   [entity entity-id-or-query prefix dashboard-template show]
-  {entity Entity
-   show   Show
-   prefix Prefix
-   dashboard-template   DashboardTemplate}
+  {entity             Entity
+   entity-id-or-query string?
```

Comment:
```suggestion
   entity-id-or-query ms/NonBlankString
```

I think the rule should be: on the api layer prefer the ms/* vars (since they have error api messages on them), then if one exists a keyword spec `:string` and when there isn't a keyword for it, to use the function, like with `pos-int?`.

Reply:
Agree, I was just too eager to finally see something and that place was in the way. 😁 Fixed all of them.